tools/appveyor.psm1

```diff
@@ -474,6 +475,19 @@ function Invoke-AppveyorFinish
             $preReleaseVersion = "$previewPrefix-$previewLabel.$env:APPVEYOR_BUILD_NUMBER"
         }
 
+        # Smoke Test MSI installer
+        Write-Verbose "Smoke-Testing MSI installer" -Verbose
+        $msi = $artifacts | Where-Object { $_.EndsWith(".msi") }
+        $msiLog = Join-Path (Get-Location) 'msilog.txt'
+        $msiExecProcess = Start-Process msiexec.exe -Wait -ArgumentList "/I $msi /quiet /l*vx $msiLog" -NoNewWindow -PassThru
+        if ($msiExecProcess.ExitCode -ne 0)
+        {
+            Push-AppveyorArtifact msiLog.txt
+            $exitCode = $msiExecProcess.ExitCode
+            throw "MSI installer failed and returned error code $exitCode. MSI Log was uploaded as artifact."
```

Comment:
Perhaps just call Add-Appveyor test with the results?

see https://github.com/PowerShell/DscResource.Tests/blob/f9e3dfd3eb9d7b6a9225e2cfba68b79ffe1cab36/AppVeyor.psm1#L459

Reply:
I have tried this in a test branch but although the test shows up as red, the build is still green: https://ci.appveyor.com/project/bergmeister/powershell/build/v6.1.0-preview.281/tests

Reply:
yup, I see and the DSC code solves this by throwing...

Reply:
No sure about the fact that it actually 'solves' it because throwing still does not make the build red. See [here](https://ci.appveyor.com/project/bergmeister/powershell/build/v6.1.0-preview.282/tests)
Even uploading a failed test result and adding a throw statement in my 2nd branch [here](https://github.com/bergmeister/PowerShell/commits/MsiInstallerSmokeTest2) does not make the build red....
![image](https://user-images.githubusercontent.com/9250262/35945537-765128a8-0c58-11e8-93ff-223ad1552aec.png)

Reply:
I think that's because we are in the finish.

Reply:
Why can this not be a Pester test that requires admin?

Reply:
Because you build all your packages in the finish step as far as I can tell.

Reply:
Thanks... forgot about that....